include/exadg/operators/generalized_laplace_operator.h

```diff
@@ -0,0 +1,781 @@
+
+    return flags;
+  }
+
+  /**
+   * Returns mapping flags required for the kernel evaluation.
+   */
+  static MappingFlags
+  get_mapping_flags(bool const compute_interior_face_integrals,
+                    bool const compute_boundary_face_integrals)
+  {
+    MappingFlags flags;
+
+    flags.cells =
+      dealii::update_JxW_values | dealii::update_gradients | dealii::update_quadrature_points;
+    if(compute_interior_face_integrals)
+      flags.inner_faces = dealii::update_JxW_values | dealii::update_gradients |
+                          dealii::update_normal_vectors | dealii::update_quadrature_points;
+    if(compute_boundary_face_integrals)
+      flags.boundary_faces = dealii::update_JxW_values | dealii::update_gradients |
+                             dealii::update_normal_vectors | dealii::update_quadrature_points;
+
+    return flags;
+  }
+
+  /**
+   * Calculates and returns the `flux` inside the cell integral. This flux should be weighted with
+   * the gradient of the shape functions.
+   */
+  static inline DEAL_II_ALWAYS_INLINE //
+    GradientType
+    get_volume_flux(GradientType const & gradient, CoefficientType const & coefficient)
+  {
+    return coeff_mult(coefficient, gradient);
+  }
+
+  /**
+   * Calculates and returns the `flux` on the faces which should be weighted with the gradient of
+   * the shape functions.
+   */
+  static inline DEAL_II_ALWAYS_INLINE //
+    GradientType
+    calculate_gradient_flux(ValueType const &       value_m,
+                            ValueType const &       value_p,
+                            Vector const &          normal,
+                            CoefficientType const & coefficient)
+  {
+    ValueType const    jump_value  = value_m - value_p;
+    GradientType const jump_tensor = outer_product(jump_value, normal);
+
+    return -0.5 * coeff_mult(coefficient, jump_tensor);
+  }
+
+  /**
+   * Calculates and returns the `flux` on the faces which should be weighted with the normal
+   * derivative of the shape functions.
+   *
+   * Weighting with the normal derivative (gradient times face normal) is mathematically only
+   * possible if the coefficient is non-coupling type. So, this is asserted inside the function
+   * body.
+   */
+  static inline DEAL_II_ALWAYS_INLINE //
+    ValueType
+    calculate_normal_derivative_flux(ValueType const &       value_m,
+                                     ValueType const &       value_p,
+                                     CoefficientType const & coefficient)
+  {
+    {Assert(coefficient_is_scalar,
+            dealii::ExcMessage("Normal derivative flux only makes"
+                               "sense with scalar coefficients."))}
+
+    ValueType const jump_value = value_m - value_p;
+
+    return -0.5 * coefficient * jump_value;
+  }
+
+  /**
+   * Calculates and returns the `flux` on the faces which should be weighted with the shape
+   * function values.
```

Comment:
It is not clear to me why the documentation of these functions should explain "weighted by ...". Implementation of the weak form is in my opinion the responsibility of the `Operator`.

Reply:
I wrote the documentation like this to make it clear where the name of the function comes from. Because that is our motivation to name this function `get_value_flux()`. The caller shall multiply the return value with the shape function values. The naming of the function is not obvious from its body, which for example also works with gradients here. I just wanted to prevent any confusion caused by the function name.